src/app/dts/dts.component.ts

```diff
@@ -24,27 +38,61 @@ export class DtsComponent {
     { prop: 'createdAt', name: 'Created'}
   ];
 
-  constructor(private dtsService: DtsService) {
+  constructor(private dtsService: DtsService, private formBuilder: FormBuilder) {
+    this.dtsForm = this.formBuilder.group({
+      templateKey: new FormControl()
+    });
+
+    this.rows = this.dtsService.getTemplates('enrollment');
+
+    this.onValueChanges();
   }
 
   /**
-   * Populates the data grid with a list of templates.
-   * @param documentType indicates type of document
+   * Filters templates by the user entered template key
    */
-  getTemplates(documentType?: string): void {
-    this.rows = this.dtsService.getTemplates(documentType);
+  onValueChanges(): void {
+    this.dtsForm.valueChanges.subscribe(val => {
+      this.rows = this.dtsService.getTemplates('enrollment')
```

Comment:
This looks like a good opportunity for using the `switchMap` operator in Rxjs. Something along those lines:
```
this.rows = this.dtsForm.valueChanges.pipe(
       switchMap(() => this.dtsService.getTemplates('enrollment')),
        map(templates => templates.filter( (template) => {
            return template.templateKey.toLowerCase().includes(val.templateKey.toLowerCase());
        })));
```

Reply:
I see what you're trying to do here, and I can investigate looking into swtichMap.  My code is currently written where it's subscribing to form changes which are stored in val.  I understand the recommendation is "something along those lines..." and it would need some rework as form changes ("val") doesn't exist.

Reply:
Yes. `val` could be read from `switchMap`like so:
```this.rows = this.dtsForm.valueChanges.pipe(
       switchMap((val) => this.dtsService.getTemplates('enrollment')),
```
then it could be stored as a component property for use in the `map` opearator.
The idea is to have just one subscription and one data stream instead of two.

Reply:
For now, I removed the Template Key Form Field so this story can get completed.  Spending too much time on this, and may eventually change the field to a type ahead control which updates the data grid when a template key is selected which would a different design where the grid updates as the user types.

Reply:
ok